packages/common-utils/src/renderChartConfig.ts

```diff
@@ -893,6 +898,21 @@ function renderFill(
   return undefined;
 }
 
+function renderDeltaExpression(chartConfig: ChartConfigWithOptDateRange) {
+  const interval =
+    chartConfig.granularity === 'auto' && Array.isArray(chartConfig.dateRange)
+      ? convertDateRangeToGranularityString(chartConfig.dateRange, 60)
+      : chartConfig.granularity;
+  const intervalInSeconds = convertGranularityToSeconds(interval ?? '');
+
+  const valueDiff = `(argMax(Value, ${chartConfig.timestampValueExpression}) - argMin(Value, ${chartConfig.timestampValueExpression}))`;
+  const timeDiffInSeconds = `date_diff('second', min(toDateTime(${chartConfig.timestampValueExpression})), max(toDateTime(${chartConfig.timestampValueExpression})))`;
+
+  // Prevent division by zero, if timeDiffInSeconds is 0, return 0
+  // The delta is extrapolated to the bucket interval, to match prometheus delta() behavior
+  return `IF(${timeDiffInSeconds} > 0, ${valueDiff} * ${intervalInSeconds} / ${timeDiffInSeconds}, 0)`;
```

Comment:
While it might be impossible under the otel (and any sane) schema to have a `NULL` date, we do allow users to define their own timestamp value expression. Thus it's possible that a `NULL` timestamp could happen. I think the query will fail in that case because the CH resolves the parameter values to the `IF` function eagerly - e.g. it's actually doing the math before it evaluates the `IF` condition.

I ran into this with histograms and had to use the `short_circuit_function_evaluation` setting to force the SQL engine to lazily eval the params to `IF`. I could also agree with making a `NOT NULL` an assumption on the timestamp field throughout the app and maybe validating that during source setup.

Reply:
Not entirely sure.